modules/mysql/mysql.go

```diff
@@ -0,0 +1,90 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+)
+
+const rootUser = "root"
+const defaultUser = "test"
+const defaultPassword = "test"
+const defaultDatabaseName = "test"
+
+// MySQLContainer represents the MySQL container type used in the module
+type MySQLContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+type MySQLContainerOption func(req *testcontainers.ContainerRequest)
+
+// StartContainer creates an instance of the MySQL container type
+func StartContainer(ctx context.Context, image string, opts ...MySQLContainerOption) (*MySQLContainer, error) {
```

Comment:
Do we still provide the image as part of the start function, or do we advocate for using a default + WithImage option?

Reply:
for consistency in tc-go, I'll add WithImage